homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +85,53 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+# Binary sensor service map
+BINARY_SENSOR_SERVICE_MAP = {
+    DEVICE_CLASS_CO2:
+        (SERV_CARBON_DIOXIDE_SENSOR, CHAR_CARBON_DIOXIDE_DETECTED),
+    DEVICE_CLASS_GAS:
+        (SERV_CARBON_MONOXIDE_SENSOR, CHAR_CARBON_MONOXIDE_DETECTED),
+    DEVICE_CLASS_MOISTURE:
+        (SERV_LEAK_SENSOR, CHAR_LEAK_DETECTED),
+    DEVICE_CLASS_MOTION:
+        (SERV_MOTION_SENSOR, CHAR_MOTION_DETECTED),
+    DEVICE_CLASS_OCCUPANCY:
+        (SERV_OCCUPANCY_SENSOR, CHAR_OCCUPANCY_DETECTED),
+    DEVICE_CLASS_OPENING:
+        (SERV_CONTACT_SENSOR, CHAR_CONTACT_SENSOR_STATE),
+    DEVICE_CLASS_SMOKE:
+        (SERV_SMOKE_SENSOR, CHAR_SMOKE_DETECTED)}
```

Comment:
Is that an option?
```py
BINARY_SENSOR_SERVICE_MAP = {
    DEVICE_CLASS_CO2: (SERV_CARBON_DIOXIDE_SENSOR,
                       CHAR_CARBON_DIOXIDE_DETECTED),
    DEVICE_CLASS_GAS: (SERV_CARBON_MONOXIDE_SENSOR,
                       CHAR_CARBON_MONOXIDE_DETECTED),
    DEVICE_CLASS_MOISTURE: (SERV_LEAK_SENSOR, CHAR_LEAK_DETECTED),
    DEVICE_CLASS_MOTION: (SERV_MOTION_SENSOR, CHAR_MOTION_DETECTED),
    DEVICE_CLASS_OCCUPANCY: (SERV_OCCUPANCY_SENSOR, CHAR_OCCUPANCY_DETECTED),
    DEVICE_CLASS_OPENING: (SERV_CONTACT_SENSOR, CHAR_CONTACT_SENSOR_STATE),
    DEVICE_CLASS_SMOKE: (SERV_SMOKE_SENSOR, CHAR_SMOKE_DETECTED)}
```

Reply:
Can you move the `BINARY_SENSOR_SERVICE_MAP` above the `TemperatureSensor` class. All constants should be defined belove the imports.

Reply:
done